tools/crm_mon.c

```diff
@@ -2807,18 +2853,363 @@ print_failed_actions(FILE *stream, pe_working_set_t *data_set)
+                    event->origin, event->client);
+            if (event->delegate) {
+                fprintf(stream, " delegate=\"%s\"", event->delegate);
+            }
+            switch(event->state) {
+                case st_done:
+                case st_failed:
+                    fprintf(stream, " completed=\"%s\"", run_at_s?run_at_s:"");
+                    break;
+                default:
+                    fprintf(stream, " state=\"pending\"");
+            }
+            fprintf(stream, " />\n");
+            break;
+
         case mon_output_plain:
         case mon_output_console:
-            print_as("\n");
+            switch(event->state) {
+                case st_done:
+                    print_as("* %s of %s successful: delegate=%s, client=%s, origin=%s,\n"
+                             "    %s='%s'\n",
+                             action_s, event->target,
+                             event->delegate ? event->delegate : "",
+                             event->client, event->origin,
+                             ((!fence_full_history) && (output_format != mon_output_xml))?
+                             "completion-last-successful":"completed",
+                             run_at_s?run_at_s:"");
+                    break;
+                case st_failed:
+                    print_as("* %s of %s failed: delegate=%s, client=%s, origin=%s,\n"
+                             "    %s='%s'\n",
+                             action_s, event->target,
+                             event->delegate ? event->delegate : "",
+                             event->client, event->origin,
+                             ((!fence_full_history) && (output_format != mon_output_xml))?
+                             "completion-last-failed":"completed",
+                             run_at_s?run_at_s:"");
+                    break;
+                default:
+                    print_as("* %s of %s pending: client=%s, origin=%s\n",
+                             action_s, event->target,
+                             event->client, event->origin);
+            }
+            break;
+
+        default:
+            /* no support for fence history for other formats so far */
+            break;
+    }
+
+    free(action_s);
+}
+
+/*!
+ * \internal
+ * \brief Print a section for failed stonith actions
+ *
+ * \param[in] stream     File stream to display output to
+ * \param[in] history    List of stonith actions
+ *
+ */
+static void
+print_failed_stonith_actions(FILE *stream, stonith_history_t *history)
+{
+    stonith_history_t *hp;
+
+    if (!history) {
+        return;
+    }
+    for (hp = history; hp; hp = hp->next) {
+        if (hp->state == st_failed) {
+            break;
+        }
+    }
+    if (!hp) {
+        return;
+    }
+
+    /* Print section heading */
+    switch (output_format) {
+        case mon_output_plain:
+        case mon_output_console:
+            print_as("\nFailed Fencing Actions:\n");
```

Comment:
Probably - was just hesitating as scripts existing out there might match on the "Failed Actions".
Other approach to get around this would have been to just stay with "Failed Actions" and add the failed fence actions to it. But then somebody parsing that section automatically would again fail.

Reply:
Don't worry about scripts -- they should be using the XML. Renaming it unconditionally is fine.